x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/flyout/use_refetch_by_scope.tsx

```diff
@@ -22,18 +22,17 @@ export interface UseRefetchScopeQueryParams {
  * Hook to refetch data within specified scope
  */
 export const useRefetchByScope = ({ scopeId }: UseRefetchScopeQueryParams) => {
-  const getGlobalQueries = inputsSelectors.globalQuery();
-  const getTimelineQuery = inputsSelectors.timelineQueryByIdSelector();
+  const getGlobalQueries = useMemo(() => inputsSelectors.globalQuery(), []);
+  const getTimelineQuery = useMemo(() => inputsSelectors.timelineQueryByIdSelector(), []);
   const { globalQuery, timelineQuery } = useDeepEqualSelector((state) => ({
     globalQuery: getGlobalQueries(state),
     timelineQuery: getTimelineQuery(state, scopeId),
   }));
 
-  const refetchQuery = (newQueries: inputsModel.GlobalQuery[]) => {
-    newQueries.forEach((q) => q.refetch && (q.refetch as inputsModel.Refetch)());
-  };
-
   const refetchAll = useCallback(() => {
+    const refetchQuery = (newQueries: inputsModel.GlobalQuery[]) => {
```

Comment:
just a note, I see this method in a couple of places, do you see any value extracting it?

Reply:
Yes there are these instances, and an order of magnitude more named just “refetch”. It should be defined and managed in one place ha, but that’s a pretty large change id prefer to do separately, but agree we should do it.